### PR TITLE
[DO NOT MERGE] Experimenting with Flow

### PR DESCRIPTION
**Pull request by zach-klippenstein**
Check out the individual commits.

### CODE REVIEW

File: kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunnerViewModel.kt [hunk 1]
```diff
@@ -65,28 +68,20 @@ internal class WorkflowRunnerViewModel<OutputT : Any>(
...
-    // Buffer on backpressure so outputs don't get lost.
-    updates.toFlowable(BUFFER)
-        .filter { it.output != null }
-        .map { it.output!! }
-        // DON'T replay, outputs are events.
-        .publish()
-        // Subscribe upstream immediately so we immediately start getting notified about outputs.
-        // If [renderings] triggers the upstream subscription before we do, if the workflow emits
-        // an output immediately we might not see it.
-        .autoConnect(0) { subs.add(it) }
```
Comment (zach-klippenstein) [inline kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunnerViewModel.kt]: @rjrjr see?

File: kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunnerViewModel.kt [hunk 2]
```diff
@@ -65,28 +68,20 @@ internal class WorkflowRunnerViewModel<OutputT : Any>(
...
   private val subs = CompositeDisposable()
 
-  @Suppress("EXPERIMENTAL_API_USAGE")
-  private val updates =
-    workflowUpdates.updates.asObservable(Dispatchers.Unconfined)
-        .doOnNext { lastSnapshot = it.snapshot }
-        .replay(1)
-        .autoConnect(1) { subs.add(it) }
+  init {
+    subs.add(workflowHost.renderingsAndSnapshots
```
Comment (rjrjr) [inline kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunnerViewModel.kt]: Sketchy to do this in the init block, but I'm pretty sure we can do better.

Comment (zach-klippenstein) [reply kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunnerViewModel.kt]: See the latest commit.

Comment (zach-klippenstein) [reply kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunnerViewModel.kt]: But we could also give it an explicit start method, same thing.

**Comment by zach-klippenstein** [comment]
Superseded by #426.

**Pull request closed without merging**